agent/lib/kontena/workers/network_create_worker.rb

```diff
@@ -0,0 +1,64 @@
+
+module Kontena::Workers
+  class NetworkCreateWorker
+    include Celluloid
+    include Celluloid::Notifications
+    include Kontena::Logging
+
+
+    def initialize
+      @network_started = false
+      @ipam_started = false
+      subscribe('network_adapter:start', :network_started)
+      subscribe('ipam:start', :ipam_started)
+      info 'initialized'
+    end
+
+    def network_started(topic, data)
+      debug 'network_started'
+      @network_started = true
+      ensure_default_network
+    end
+
+    def ipam_started(topic, data)
+      debug 'ipam_started'
+      @ipam_started = true
+      ensure_default_network
+    end
+
+    
+    def create_network(name, driver, ipam_driver, subnet, ip_range = nil)
+      opts = {
+        'Driver': driver,
```

Comment:
yes.
```
2.3.1 :001 > opts = {'Driver':'foo'}
 => {:Driver=>"foo"}
```
IMO it's bit more readable this way since the target for the data is Docker remote API with json payload.